omod/src/main/webapp/resources/scripts/jquery-ui/js/openmrsSearch.js

```diff
@@ -546,26 +546,38 @@ function OpenmrsSearch(div, showIncludeVoided, searchHandler, selectionHandler,
             });
 
             // Browser back button support, for preserve data
-            if (lastSearchParams !== null) {
-                $j('#inputNode').val(lastSearchParams.lastSearchText);
-                if (lastSearchParams.includeVoided == 1) {
+            var lastSearchText = decodeURIComponent((new RegExp('[?|&]' + 'lastSearchText' 
```

Comment:
Ah, I haven't used history.pushState. I see how to do it now! You can learn so much from even such a little fix. Thanks for your help.